src/Api/Parser/EventParsingIterator.php

```diff
@@ -69,36 +69,43 @@ private function parseEvent(array $event)
             if ($event['headers'][':message-type'] === 'error') {
                 return $this->parseError($event);
             }
+
             if ($event['headers'][':message-type'] !== 'event') {
                 throw new ParserException('Failed to parse unknown message type.');
             }
         }
 
-        if (empty($event['headers'][':event-type'])) {
+        $eventType = $event['headers'][':event-type'] ?? null;
+        if (is_null($eventType)) {
             throw new ParserException('Failed to parse without event type.');
         }
-        $eventShape = $this->shape->getMember($event['headers'][':event-type']);
 
+        $eventShape = $this->shape->getMember($eventType);
+        $eventPayload = $event['payload'];
         $parsedEvent = [];
-        foreach ($eventShape['members'] as $shape => $details) {
-            if (!empty($details['eventpayload'])) {
-                $payloadShape = $eventShape->getMember($shape);
-                if ($payloadShape['type'] === 'blob') {
-                    $parsedEvent[$shape] = $event['payload'];
+        foreach ($eventShape['members'] as $eventShapeMemberKey => $eventShapeMemberProps) {
+            $memberShape = $eventShape->getMember($eventShapeMemberKey);
+            if (!empty($eventShapeMemberProps['eventheader'])) {
+                $parsedEvent[$eventShapeMemberKey] = $event['headers'][$eventShapeMemberKey];
+            } elseif (!empty($eventShapeMemberProps['eventpayload'])) {
+                if ($memberShape['type'] === 'blob') {
```

Comment:
pretty sure you can use a `getType` call here

Reply:
Done!